### PR TITLE
feat: add the ability to filter apps based on clusters

### PR DESCRIPTION
**Pull request by my-git9**
Add the ability to filter apps based on clusters

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by my-git9** [comment]
The test result:
```shell
$  ./dist/argocd app list -c https://10.29.8.20:16443
NAME                  CLUSTER                   NAMESPACE    PROJECT  STATUS     HEALTH   SYNCPOLICY  CONDITIONS                REPO                                                  PATH       TARGET
argocd/guestbookproj  https://10.29.8.20:16443  kube-public  test     OutOfSync  Healthy  <none>      SharedResourceWarning(2)  https://gitlab.daocloud.cn/xin.li/argocd-example.git  guestbook
$
$  ./dist/argocd app list --cluster https://kubernetes.default.svc
NAME                   CLUSTER                         NAMESPACE    PROJECT  STATUS     HEALTH       SYNCPOLICY  CONDITIONS  REPO                                                  PATH            TARGET
argocd/apps            https://kubernetes.default.svc  default      default  OutOfSync  Progressing  <none>      <none>      https://gitlab.daocloud.cn/xin.li/argocd-example.git  apps            HEAD
argocd/guestbook       https://kubernetes.default.svc  default      default  Synced     Healthy      Auto        <none>      https://gitlab.daocloud.cn/xin.li/argocd-example.git  guestbook       HEAD
argocd/guestbook-helm  https://kubernetes.default.svc  default      default  Synced     Healthy      <none>      <none>      https://gitlab.daocloud.cn/xin.li/argocd-example.git  helm-guestbook  HEAD
argocd/guestbook-pub   https://kubernetes.default.svc  kube-public  default  Synced     Healthy      <none>      <none>      https://gitlab.daocloud.cn/xin.li/argocd-example.git  guestbook
argocd/test            https://kubernetes.default.svc  default      default  Synced     Healthy      <none>      <none>      https://gitlab.daocloud.cn/xin.li/argocd-example.git  helm-hooks

```

**Comment by my-git9** [comment]
Hi @jannfis, Can you review this PR? Thanks

File: util/argo/argo.go [hunk 1]
```diff
@@ -75,6 +75,20 @@ func FilterByRepo(apps []argoappv1.Application, repo string) []argoappv1.Applica
...
 }
 
+// FilterByCluster returns an application
+func FilterByCluster(apps []argoappv1.Application, cluster string) []argoappv1.Application {
+	if cluster == "" {
+		return apps
+	}
+	items := make([]argoappv1.Application, 0)
+	for i := 0; i < len(apps); i++ {
+		if apps[i].Spec.Destination.Server == cluster {
```
Comment (ashutosh16) [inline util/argo/argo.go:85]: Do we need to support filter apps by cluster names? What if the user provides a cluster name instead of url?

Comment (my-git9) [reply util/argo/argo.go:85]: Sure, I can make another PR to do the job

Comment (ashutosh16) [reply util/argo/argo.go:85]: I think you can add one more condition here to check cluster by its symbolic name
```
if apps[i].Spec.Destination.Server == cluster ||  apps[i].Spec.Destination.Name == cluster {
```

Comment (my-git9) [reply util/argo/argo.go:85]: Good idea!

**Review by ashutosh16 — APPROVED** [review]
LGTM

File: cmd/argocd/commands/app.go [hunk 2]
```diff
@@ -1293,6 +1296,7 @@ func NewApplicationListCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	command.Flags().StringArrayVarP(&projects, "project", "p", []string{}, "Filter by project name")
 	command.Flags().StringVarP(&repo, "repo", "r", "", "List apps by source repo URL")
 	command.Flags().StringVarP(&appNamespace, "app-namespace", "N", "", "Only list applications in namespace")
+	command.Flags().StringVarP(&cluster, "cluster", "c", "", "List apps by cluster")
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:1299]: Thanks for this @my-git9! Can you update this string to clarify that either a URL or cluster name is acceptable?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/app.go (hunk 2 above)
Comment (my-git9) [reply cmd/argocd/commands/app.go]: Updated.Do you think this is ok?

**Review by crenshaw-dev — APPROVED** [review]
Thanks @my-git9!

**Pull request merged**